### PR TITLE
Issue 45

### PR DESCRIPTION
**Pull request by Jdsleppy**
### Changes:
- Type annotations to the following (no mypy errors other than missing stubs):
    - course.py
    - lab.py
    - laundry.py
    - library.py
- Fix `lab_test.test_make_status`

### References:
- Issue #45 

I was planning on annotating the entire project before making a pull request, but I'm encountering a painful amount of merge conflicts and want to get this in sooner than later.  I'm happy to make any changes as needed.

### CODE REVIEW

File: PittAPI/laundry.py [hunk 1]
```diff
@@ -88,34 +88,34 @@ def get_status_detailed(building_name: str) -> List[Dict[str,str]]:
...
     for status_string in resp_split:
         if "machine" not in status_string:
             continue
         machine_name = status_string[:status_string.index('=')].replace('Status', '') #type: str
         status_string = status_string[status_string.index('=') + 1:].strip()
 
-        old_machine_split = status_string.split("\n") #type: List[str]
-        old_machine_split[0] += machine_name
+        status_lines = status_string.split("\n") #type: List[str]
+        status_lines[0] += machine_name
```
Comment (Jdsleppy) [inline PittAPI/laundry.py]: This is another instance where I normally wouldn't have renamed this variable, but my changes conflicted with someone else who renamed it, and I think the name `status_lines` is clearer.

File: PittAPI/course.py [hunk 2]
```diff
@@ -20,8 +20,8 @@
 
 import requests
 import re
-from typing import List, Dict, Tuple, Any
-from bs4 import BeautifulSoup, SoupStrainer
+from typing import Any, Dict, List, Tuple
+from bs4 import BeautifulSoup, SoupStrainer, Tag, ResultSet
```
Comment (Jdsleppy) [inline PittAPI/course.py]: Rebase conflicts... I just tend to put things in alphabetical order.

File: PittAPI/lab.py [hunk 3]
```diff
@@ -19,7 +19,6 @@
 
 import requests
 from bs4 import BeautifulSoup
-
```
Comment (Jdsleppy) [inline PittAPI/lab.py]: Normally I wouldn't impose my style on the project, but I already had this committed and this stuck around after rebasing...

File: tests/lab_test.py [hunk 4]
```diff
@@ -107,8 +107,8 @@ def test_make_status(self):
         self.assertEqual(open[keys[0]], 'open')
 
         for key in keys[1:]:
-            self.assertEqual(closed[key], 0)
-            self.assertEqual(open[key], 1)
+            self.assertEqual(closed[key], '0')
+            self.assertEqual(open[key], '1')
```
Comment (Jdsleppy) [inline tests/lab_test.py]: These are all cast to strings here: https://github.com/Pitt-CSC/PittAPI/pull/82/files#diff-1c156c9a7cf25c89ef4fd8323d5c4d35L66

**Comment by RitwikGupta** [comment]
This is awesome! I will merge this PR, and then you can continue with this to avoid conflicts :) Thank you!

**Pull request merged**